lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -278,9 +314,15 @@ async sub transform_transaction {
     my $transaction;
 
     try {
-        my $receipt = await $self->rpc_client->get_transaction_receipt($decoded_transaction->{hash});
+        my $receipt;
+        my $gas;
+        if ($decoded_transaction->{input} ne '0x') {
+            $receipt = await $self->rpc_client->get_transaction_receipt($decoded_transaction->{hash});
+            $gas     = $receipt->{gasUsed};
+        }
+
         # fee = gas * gasPrice
-        my $fee = Math::BigFloat->from_hex($receipt->{gasUsed} // $decoded_transaction->{gas})->bmul($decoded_transaction->{gasPrice});
+        my $fee = Math::BigFloat->from_hex($gas // $decoded_transaction->{gas})->bmul($decoded_transaction->{gasPrice});
```

Comment:
Instead of this better to do `my $gas = $decoded_transaction->{gas}` on line 318 and this will just become `Math::BigFloat->from_hex($gas)->bmul($decoded_transaction->{gasPrice})`